perma_web/perma/tasks.py

```diff
@@ -61,7 +60,87 @@
 AFTER_LOAD_TIMEOUT = 30 # seconds to allow page to keep loading additional resources after onLoad event fires
 VALID_FAVICON_MIME_TYPES = {'image/png', 'image/gif', 'image/jpg', 'image/jpeg', 'image/x-icon', 'image/vnd.microsoft.icon', 'image/ico'}
 
-### HELPERS ###
+### THREAD HELPERS ###
+
+def run_capture_component(thread_list, func, **kwargs):
+    """
+       Run in a new thread, unless THREADED_CAPTURES is False (for debugging)
+    """
+    if settings.THREADED_CAPTURES:
+        add_thread(thread_list, func, **kwargs)
+    else:
+        func(**kwargs["kwargs"])
+
+def add_thread(thread_list, target, **kwargs):
+    if not isinstance(target, threading.Thread):
+        target = threading.Thread(target=target, **kwargs)
+    target.start()
+    thread_list.append(target)
+    return target
+
+def safe_save_fields(instance, **kwargs):
+    """
+        Update and save the given fields for a model instance.
+        Use update_fields so we won't step on changes to other fields made in another thread.
+    """
+    for key, val in kwargs.items():
+        setattr(instance, key, val)
+    instance.save(update_fields=kwargs.keys())
+
+def get_url(url, thread_list, proxy_address):
+    """
+        Get a url, via proxied python requests.get(), in a way that is interruptable from other threads.
+        Blocks calling thread. Perma assumes only run from sub-threads, and settings.THREADED_CAPTURES = True.
+        Unexpected blocking may result otherwise.
```

Comment:
Yes.

I was originally hoping to wrap the get requests for favicons and robots.txt so that they could done synchronously, but it got pretty ugly. So I gave up on that, at least temporarily. With THREADED_CAPTURES = False, this function is used for favicons and robots.txt, but the main thread is blocked, so things at least happen one at a time. Given that -that- didn't work out, and given that the media tags requests are still 100% asynchronous regardless of how THREADED_CAPTURES is set.... I think I'll remove all that from the pull request, and save that for another day. 

Re: the comment. The original comment for this function said, "Helper function to get a url, via proxied requests.get(), in a way that is interruptable from other threads. This should only be run from sub-threads." I was trying to alter that comment to explicitly state that, if called from the main thread or if THREADED_CAPTURES = False for debugging, it will block.